maestro-test/src/test/kotlin/maestro/test/IntegrationTest.kt

```diff
@@ -2738,6 +2747,79 @@ class IntegrationTest {
                 Event.InputText("bar"),
             )
         )
+        assertThat(receivedLogs).containsExactly("Hello from GraalJS")
+    }
+
+    @Test
+    fun `Case 102 - GraalJs dangerous config fails by default`() {
+        // given
+
+        val commands = readCommands("102_graaljs_dangerous")
+        val driver = driver { }
+        val thrownErrors = mutableListOf<KClass<out Throwable>>()
+
+        // when
+        Maestro(driver).use {
+            orchestra(
+                it,
+                onCommandFailed = { _, _, e ->
+                    thrownErrors += e::class
+                    Orchestra.ErrorResolution.CONTINUE
+                },
+            ).runFlow(commands)
+        }
+
+        // then
+        assertThat(thrownErrors).containsExactly(PolyglotException::class)
+    }
+
+    @Test
+    fun `Case 102 - GraalJs dangerous config fails with maestro env`() {
+        // given
+
+        val commands = readCommands("102_graaljs_dangerous_env").withEnv(
+            mapOf(
+                "MAESTRO_CLI_DANGEROUS_GRAALJS_ALLOW_HOST_ACCESS" to "1",
+                "MAESTRO_CLI_DANGEROUS_GRAALJS_ALLOW_CLASS_LOOKUP" to "y",
+            )
+        )
+        val driver = driver { }
+        val thrownErrors = mutableListOf<KClass<out Throwable>>()
+
+        // when
+        Maestro(driver).use {
+            orchestra(
+                it,
+                onCommandFailed = { _, _, e ->
+                    thrownErrors += e::class
+                    Orchestra.ErrorResolution.CONTINUE
+                },
+            ).runFlow(commands)
+        }
+
+        // then
+        assertThat(thrownErrors).containsExactly(PolyglotException::class)
+    }
+
+    @Test @Disabled("Need to find a way to mock System.env. Remove this and use the 'integration_102_graaljs_access_enabled' run config.")
+    fun `Case 102 - GraalJs dangerous config enabled`() {
```

Comment:
That's indeed a bummer 🫤

Long term, I'd prefer for all classes in this project to explicitly require all dependencies (such as env vars, file system, etc.) in constructors. For example if a class required env vars, it would require a  `Map<String, String` in its constructor.

That'd be a big refactor though, and hard to do without breakages given our insufficient test coverage.

Reply:
I know.
We could simply also declare a small separate helper function that can easily be stubbed for this purpose.

Reply:
I see it's been done with the introduction of the `object Env` and `MockK` library to mock that object. Thanks!